src/trajectory.jl

```diff
@@ -400,11 +389,13 @@ $(TYPEDFIELDS)
 # References
 1. Betancourt, M. (2017). A Conceptual Introduction to Hamiltonian Monte Carlo. [arXiv preprint arXiv:1701.02434](https://arxiv.org/abs/1701.02434).
 """
-struct GeneralisedNoUTurn{T<:AbstractVector{<:Real}} <: AbstractTerminationCriterion
-    "Integral or sum of momenta along the integration path."
-    rho::T
+@with_kw struct GeneralisedNoUTurn{F<:Real} <: DynamicTerminationCriterion
+    max_depth::Int=10
+    Δ_max::F=1000
 end
 
+const NoUTurn = GeneralisedNoUTurn
```

Comment:
I want to make the transition of calling the old ones `ClassificNoUTurn` and new ones just `NoUTurn` and `StrictNoUTurn`, i.e. by default we refer to the generalised ones if not specified.

Reply:
Perhaps it would make sense to go the other way around, then? Define `NoUTurn` and then make `GeneralisedNoUTurn` the alias so that users' legacy code works?

Reply:
There were some debates between me and Hong on if we want to make this transition, and we ended up with this compromise. So with the PR, the recommended way is still the previous version. And we want to use the shorter version privately for a while before making the transition.